tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,676 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <pup.h>
+#include <random>
+
+#include "DataStructures/DataBox/Prefixes.hpp" // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/DampedHarmonicGaugeQuantities.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+template <typename DataType, size_t SpatialDim, typename Frame>
+void test_weight_functions(const DataVector& used_for_size) noexcept {
+  // weight_function
+  pypp::check_with_random_values<1>(
+      static_cast<Scalar<DataType> (*)(
+        const tnsr::I<DataType, SpatialDim, Frame>&,
+        const double&, const double&)>(
+          &::GeneralizedHarmonic::DampedHarmonicGaugeHelpers::
+            weight_function<SpatialDim, Frame, DataType>),
+      "TestFunctions", "weight_func",
+      {{{std::numeric_limits<double>::denorm_min(), 10.0}}}, used_for_size);
```

Comment:
the coordinates in `weight_func` can also be negative. The test will probably also fail if `r_max` becomes very tiny. Give bounds for every argument of the function individually, e.g.
`{{{-10.0, 10.0}, {std::numeric_limits<double>::denorm_min(), 10.0}, {0.1, 10.0}}}`